kerastuner/engine/hyperparameters.py

```diff
@@ -56,27 +56,53 @@ class Choice(HyperParameter):
 
     Args:
         name: Str. Name of parameter. Must be unique.
-        values: List of possible values.
-            Any serializable type is allowed.
+        values: List of possible values. Values must be int, float,
+            str, or bool. All values must be of the same type.
+        ordered: Whether the values passed should be considered to
+            have an ordering. This defaults to `True` for float/int
+            values. Must be `False` for any other values.
         default: Default value to return for the parameter.
             If unspecified, the default value will be:
             - None if None is one of the choices in `values`
             - The first entry in `values` otherwise.
     """
 
-    def __init__(self, name, values, default=None):
+    def __init__(self, name, values, ordered=None, default=None):
         super(Choice, self).__init__(name=name, default=default)
         if not values:
             raise ValueError('`values` must be provided.')
         self.values = values
+
+        # Type checking.
+        types = set(type(v) for v in values)
+        unsupported_types = types - {int, float, str, bool}
+        if unsupported_types:
+            raise TypeError(
+                'A `Choice` can contain only `int`, `float`, '
+                '`str`, or `bool`, found: {}'.format(unsupported_types))
+        if len(types) > 1:
+            raise TypeError(
+                'A `Choice` can contain only one type of value, '
+                'found: {}'.format(types))
```

Comment:
Please also print the list of values passed by the user

Reply:
Done